workitem/event/event_repository_blackbox_test.go

```diff
@@ -0,0 +1,98 @@
+package event_test
+
+import (
+	"testing"
+
+	"github.com/fabric8-services/fabric8-wit/account"
+	"github.com/fabric8-services/fabric8-wit/gormtestsupport"
+	"github.com/fabric8-services/fabric8-wit/resource"
+	tf "github.com/fabric8-services/fabric8-wit/test/testfixture"
+	"github.com/fabric8-services/fabric8-wit/workitem"
+	"github.com/fabric8-services/fabric8-wit/workitem/event"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type eventRepoBlackBoxTest struct {
+	gormtestsupport.DBTestSuite
+	wiRepo       workitem.WorkItemRepository
+	wiEventRepo  event.WorkItemEventRepository
+	identityRepo account.IdentityRepository
+}
+
+func TestRunEventRepoBlackBoxTest(t *testing.T) {
+	resource.Require(t, resource.Database)
+	suite.Run(t, &eventRepoBlackBoxTest{
+		DBTestSuite: gormtestsupport.NewDBTestSuite("../../config.yaml"),
+	})
+}
+
+func (s *eventRepoBlackBoxTest) SetupTest() {
+	s.DBTestSuite.SetupTest()
+	s.wiRepo = workitem.NewWorkItemRepository(s.DB)
+	s.wiEventRepo = event.NewWorkItemEventRepository(s.DB)
+	s.identityRepo = account.NewIdentityRepository(s.DB)
+}
+
+func (s *eventRepoBlackBoxTest) TestList() {
+
+	s.T().Run("empty event list", func(t *testing.T) {
+
+		fxt := tf.NewTestFixture(t, s.DB, tf.WorkItems(1))
+
+		eventList, err := s.wiEventRepo.List(s.Ctx, fxt.WorkItems[0].ID)
+		require.NoError(t, err)
+		require.Empty(t, eventList)
+	})
+
+	s.T().Run("event assignee - previous assignee nil", func(t *testing.T) {
+
+		fxt := tf.NewTestFixture(t, s.DB, tf.WorkItems(1))
+
+		assignee := []string{fxt.Identities[0].ID.String()}
+
+		fxt.WorkItems[0].Fields[workitem.SystemAssignees] = assignee
+		wiNew, err := s.wiRepo.Save(s.Ctx, fxt.WorkItems[0].SpaceID, *fxt.WorkItems[0], fxt.Identities[0].ID)
+		require.NoError(t, err)
+		require.Len(t, wiNew.Fields[workitem.SystemAssignees].([]interface{}), 1)
+		eventList, err := s.wiEventRepo.List(s.Ctx, fxt.WorkItems[0].ID)
+		require.NoError(t, err)
+		require.NotEmpty(t, eventList)
+		require.Len(t, eventList, 1)
+		assert.Equal(t, eventList[0].Name, "assigned")
+		assert.Empty(t, eventList[0].PreviousAssignees)
+		assert.Equal(t, fxt.Identities[0].Username, eventList[0].NewAssignees[0])
+	})
+
+	s.T().Run("event assignee - new assignee nil", func(t *testing.T) {
+
+		fxt := tf.NewTestFixture(t, s.DB, tf.WorkItems(1))
+
+		assignee := []string{fxt.Identities[0].ID.String()}
+
+		fxt.WorkItems[0].Fields[workitem.SystemAssignees] = assignee
+		wiNew, err := s.wiRepo.Save(s.Ctx, fxt.WorkItems[0].SpaceID, *fxt.WorkItems[0], fxt.Identities[0].ID)
+		require.NoError(t, err)
+		require.Len(t, wiNew.Fields[workitem.SystemAssignees].([]interface{}), 1)
+		eventList, err := s.wiEventRepo.List(s.Ctx, fxt.WorkItems[0].ID)
+		require.NoError(t, err)
+		require.NotEmpty(t, eventList)
+		require.Len(t, eventList, 1)
+		assert.Equal(t, eventList[0].Name, "assigned")
```

Comment:
I think we should use constants for the various supported types of event names.

Reply:
fixed 1e56c85fc928d4b342822390094df4145aa1159b